src/main/com/yetanalytics/flint/axiom/impl/validation.cljc

```diff
@@ -1,55 +1,561 @@
-(ns com.yetanalytics.flint.axiom.impl.validation)
+(ns com.yetanalytics.flint.axiom.impl.validation
+  #?(:clj  (:import [java.util BitSet])
+     :cljs (:require [goog.string :refer [format]]
+                     [goog.string.format])))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; Regex
+;; Ranges
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; 
 
-(def iri-regex*
-  #"([^<>{}\"\\|\^`\s])*")
+(defn- char->int
+  [c]
+  #?(:clj (int c) :cljs (.charCodeAt c)))
 
-(def iri-regex
-  #"<([^<>{}\"\\|\^`\s])*>")
+#?(:clj (def ^:private qmark-range [(char->int \?)]))
+#?(:clj (def ^:private uscore-range [(char->int \_)]))
+#?(:clj (def ^:private hyphen-range [(char->int \-)]))
+#?(:clj (def ^:private bslash-range [(char->int \\)]))
+#?(:clj (def ^:private percent-range [(char->int \%)]))
+#?(:clj (def ^:private open-abrac-range [(char->int \<)]))
+#?(:clj (def ^:private close-abrac-range [(char->int \>)]))
 
-(def prefix-iri-ns-regex
-  #"[A-Za-z_]([\w\-\.]*[\w\-])?")
+(def ^:private unicode-range-no-digits
+  [[(char->int \A) (char->int \Z)]
+   [(char->int \a) (char->int \z)]
+   [0x00C0 0x00D6] ; * Latin letters with diacritics + spacing modifiers
+   [0x00D8 0x00F6] ;   Excludes multiplication + division signs
+   [0x0370 0x037D] ; * Various alphabets (e.g. Greek, Cyrillic, Arabic)
+   [0x037F 0x1FFF] ;   Excludes Greek question mark
```

Comment:
Probably for the best; lol

Reply:
It's actually explicitly forbidden by the spec: https://www.w3.org/TR/sparql11-query/#rPN_CHARS_BASE

Reply:
As it should be. Related: https://twitter.com/benbjohnson/status/533848879423578112